src/span.js

```diff
@@ -249,35 +260,97 @@ export default class Span {
         }
     }
 
-    _toThrift() {
-        let tags = [];
-        for (let i = 0; i < this._tags.length; i++) {
-            let tag = this._tags[i];
-            let emptyBuffer = new Buffer(1);
-            emptyBuffer.fill(0);
-            tags.push({
-                key: tag.key,
-                tagType: 'STRING', // TODO(oibe) support multiple tag types
-                vStr: tag.value.toString(),
-                vDouble: 0,
-                vBool: false,
-                vInt16: 0,
-                vInt32: 0,
-                vInt64: 0,
-                vBinary: emptyBuffer
+    static _getTagType(tag: Tag): string {
+        if (tag.value instanceof Number) {
+            return Span._thrift.TagType.DOUBLE;
+        } else if (tag.value instanceof Boolean) {
+            return Span._thrift.TagType.BOOL;
+        } else if (tag.value instanceof Long) { //TODO(oibe) how else to recognize a long?
+            return Span._thrift.TagType.LONG;
+        } else if (tag.value instanceof Buffer) {
+            return Span._thrift.TagType.BUFFER;
+        } else {
+            return Span._thrift.TagType.STRING;
+        }
+    }
+
+    static _getThriftTags(initialTags: Array<Tag>): Array<any> {
+        let thriftTags = [];
+        for (let i = 0; i < initialTags.length; i++) {
+            let tag = initialTags[i];
+
+            let key: string = tag.key;
+            let vType: string = Span._getTagType(tag);
```

Comment:
this is doing double work of switching on the type, why not combine it in a single if?